tarsier/tag_utils.ts

```diff
@@ -506,19 +544,30 @@ function insertTags(
         const parentXPath = getElementXPath(el);
         const textNodeIndex = allTextNodes.indexOf(child) + 1;
 
-        idToTagMetadata[idNum] = {
+        const idSpan = create_tagged_span(idNum, el);
+
+        const idSpanContent = idSpan.textContent || "";
+        const idSymbol = extractIdSymbol(idSpanContent);
+        const idString = idSymbol ? `[ ${idSymbol} ${idNum} ]` : `[ ${idNum} ]`;
+
+        tagMetadataList.push({
+          tarsierID: idNum,
+          elementName: elementName,
+          elementHTML: elementHTML,
           xpath: parentXPath,
+          elementText: child.textContent?.trim() || null,
           textNodeIndex: textNodeIndex,
-        };
+          idSymbol: idSymbol,
+          idString: idString,
```

Comment:
it's not ideal to manually construct this giant thing in three places. We should instead keep track of 

```
xpath,
element,
tagElement,
textNodeIndex
```

everywhere and then in a separate outer loop we can construct the actual tagMetadataList with all of the required fields. 

Ideally we have a larger refactor here to fix a lot of the redundancy we have here, but that's probably out of scope